### PR TITLE
Tests for package_search on Registry and Portal

### PR DESCRIPTION
**Pull request by JVickery-TBS**
feat(tests): added package_search tests;

- Added test class for package_search on Registry.
- Added test class for package_search on Portal.

### CODE REVIEW

File: ckanext/canada/tests/test_search.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        self.user_lc = LocalCKAN(username=user['name'])
+        self.editor_lc = LocalCKAN(username=editor['name'])
+        self.system_lc = LocalCKAN(username=sysadmin['name'])
+        #
+        # Datasets 0, 1, 2, and 3
+        # 1 & 3 will be ready to publish
+        # 3 will not have a portal release date
+        # datasets belong to self.org
+        #
+        for _i in range(0, 4):  # 0-3
```
Comment (wardi) [inline ckanext/canada/tests/test_search.py:39]: `i` is a common loop variable, why `_i`?

Comment (JVickery-TBS) [reply ckanext/canada/tests/test_search.py:39]: @wardi I just personally use `_<var name>` for local scope. And the `_` before the variable in the loop will ignore any typing for the basic loop.

**Comment by wardi** [comment]
Looks good. These will be great to have.

Next bits of core functionality we don't have covered are requesting user accounts and working with the PD UI. Make some tickets for the backlog?

File: ckanext/canada/tests/test_search.py (hunk 1 above)
Comment (wardi) [reply ckanext/canada/tests/test_search.py:39]: in Python `_<var name>` means a name that shouldn't be imported from outside modules, so it's kind of funny looking. What do you mean about ignoring typing?

Comment (wardi) [reply ckanext/canada/tests/test_search.py]: All variables in python are local unless you don't declare them or use `nonlocal foo` or `global foo` in the function, and there is a convention of *not* prefixing variable names for scope, type or anything else. There are some conventions around capitalization, however.

**Pull request merged**